### PR TITLE
Add DatabaseVulnerabilityAssessments swagger

### PR DESCRIPTION
**Pull request by ggildin**
The put api is updated.
A new parameter is added to its body properties storageAccountAccessKey.
storageContainerSasKey isn't required anymore, but the user must to specified one of the following parameters:
storageContainerSasKey
or
storageAccountAccessKey

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by jaredmoo** [comment]
Please look at travis CI failures

**Comment by jaredmoo** [comment]
Are api-versions other than 2017-10 affected?

**Comment by ggildin** [comment]
Hi.
1. I took a look on CI failures, hope that are fixed now.
2. Since it only add an optional (exclusive or) parameter, we don't think that a new api version is required.

**Comment by lmazuel** [comment]
@ggildin @jaredmoo The new file is didn't plug into any tags, is this on purpose?

**Comment by jaredmoo** [comment]
Let me rephrase my question. Here you are adding api-version 2017-10, but does api-version 2017-03 or earlier also contain the change?

**Comment by ggildin** [comment]
Hi.
I updated the 2017-03 documentation as well.
We don't have documentation at earlier versions.

**Comment by ggildin** [comment]
@yaakoviyun

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/databaseVulnerabilityAssessments.json [hunk 1]
```diff
@@ -261,12 +271,44 @@
...
+      ],
+      "properties": {}
+    },
     "DatabaseVulnerabilityAssessment": {
       "description": "A database vulnerability assessment.",
       "type": "object",
       "allOf": [
         {
-          "$ref": "../../../common/v1/types.json#/definitions/ProxyResource"
+          "$ref": "#/definitions/ProxyResource"
```
Comment (yaakoviyun) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/databaseVulnerabilityAssessments.json]: put
-          "$ref": "../../../common/v1/types.json#/definitions/ProxyResource"

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/databaseVulnerabilityAssessments.json [hunk 2]
```diff
@@ -226,7 +228,15 @@
...
+          "x-ms-mutability": [
+            "create",
+            "update"
+          ]
+        },
+        "storageAccountAccessKey": {
+          "description": "Specifies the identifier key of the auditing storage account. If 'StorageContainerSasKey' isn't specified, storageAccountAccessKey is required.",
           "type": "string",
           "x-ms-mutability": [
             "create",
```
Comment (yaakoviyun) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/databaseVulnerabilityAssessments.json]: +            "create",
+            "update"?

**Comment by ggildin** [comment]
Hi.
1. I synced with @yaakoviyun, we don't want to exposed at frontend a new version of the api. So I removed the swagger file for 2017-10-01 versions. (The auth generated tool generated them).
2. I updated the 2017-03-01 version files.
3. After this changes no swagger file is added, so no tag updating is required.
3.  @jaredmoo - Regards Travis error - The failure is due to breaking change. Required field has been changed to be optional. As we talked in the CR, this is only a minor change and is acceptable.
Do I have any way to exclude this failure?

**Comment by ravbhatnagar** [comment]
Looks good.

**Comment by jaredmoo** [comment]
Please check travis errors:
- https://travis-ci.org/Azure/azure-rest-api-specs/jobs/396018461#L692
- https://travis-ci.org/Azure/azure-rest-api-specs/jobs/396018480#L741

**Comment by jaredmoo** [comment]
There is still model validation error: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/397699547

**Comment by ggildin** [comment]
fixed

**Comment by lmazuel** [comment]
@jaredmoo That's a preview Swagger, signed-of by ARM, and changes look good me. Once you think it's done, let me know and I merge the Pr.
Thanks!

**Review by jaredmoo — APPROVED** [review]

**Pull request merged**